pkg/environment/environment.go

```diff
@@ -20,6 +20,9 @@ const (
 	// CloudProviderConfigLocationVarName is an environment variable name. This file is available on azure cluster.
 	CloudProviderConfigLocationVarName = "AZURE_CLOUD_PROVIDER_LOCATION"
 
+	// ClientIDVarName is an environment variable which stores the client id provided through user assigned identity
+	ClientIDVarName = "AZURE_CLIENT_ID"
```

Comment:
AZURE_UMI_CLIENT_ID?

Reply:
This is env variable defined the by Azure Go SDK. So we have to use this string.
https://github.com/Azure/azure-sdk-for-go#more-authentication-details